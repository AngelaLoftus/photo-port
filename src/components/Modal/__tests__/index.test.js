import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Modal from '../'

const mockToggleModal = jest.fn();

const currentPhoto = {
    name: 'park bench',
    category: 'landscape',
    description: 'lorem ipsum',
    index: 1
};

afterEach(cleanup);

describe('Modal component', () => {
    it ('renders', () => {
        //baseline render component test
        render(<Modal 
            onClose = {mockToggleModal}
            currentPhoto={currentPhoto}
        />);
    });
    
    //snapshot test
    it('matches snpashot DOM', () => {
        const { asFragment } = render(<Modal
        onClose={mockToggleModal}
        currentPhoto={currentPhoto}
            />);

            expect(asFragment()).toMatchSnapshot();
    })
})

describe('click event', () => {
    it('calls onClose handler', () => {
        //arrange render modal
        const {getByText} = render(<Modal 
            onClose = {mockToggleModal}
            currentPhoto={currentPhoto} />);
        //act simulate click event
            fireEvent.click(getByText('Close this modal'));
        //assert expected matcher
        expect(mockToggleModal).toHaveBeenCalledTimes(1);
    })
})