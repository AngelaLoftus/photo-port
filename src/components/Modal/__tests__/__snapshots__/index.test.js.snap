// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component matches snpashot DOM 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      >
        park bench
      </h3>
      <img
        alt="current category"
        src="1.jpg"
      />
      <p>
        lorem ipsum
      </p>
      <button
        type="button"
      >
        Close this modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
